### PR TITLE
Make GenericsChecks methods static

### PR DESCRIPTION
**Pull request by msridhar**
Previously, some methods in `GenericsChecks` were static methods and some were instance methods.  We did not get much of any benefit from the instance methods, as it would be painful to create a single `GenericsChecks` object and thread it everywhere it is needed.  So, this PR just converts all `GenericsChecks` methods to be static and passes parameters as needed (which isn't too painful).

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 1]
```diff
@@ -287,21 +290,24 @@ public void checkTypeParameterNullnessForAssignability(Tree tree) {
...
       if (!isAssignmentValid) {
         reportInvalidAssignmentInstantiationError(tree, lhsType, rhsType, state, analysis);
       }
     }
   }
 
-  public void checkTypeParameterNullnessForFunctionReturnType(
-      ExpressionTree retExpr, Symbol.MethodSymbol methodSymbol) {
-    if (!config.isJSpecifyMode()) {
+  public static void checkTypeParameterNullnessForFunctionReturnType(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:305]: This is a public method, so... javadoc?

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 2]
```diff
@@ -237,7 +239,7 @@ private void reportInvalidOverridingMethodParamTypeError(
    * @return Type of the tree with preserved annotations.
    */
   @Nullable
-  private Type getTreeType(Tree tree) {
+  private static Type getTreeType(Tree tree, VisitorState state) {
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:242]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 3]
```diff
@@ -267,8 +269,9 @@ private Type getTreeType(Tree tree) {
    * @param tree the tree to check, which must be either an {@link AssignmentTree} or a {@link
    *     VariableTree}
    */
-  public void checkTypeParameterNullnessForAssignability(Tree tree) {
-    if (!config.isJSpecifyMode()) {
+  public static void checkTypeParameterNullnessForAssignability(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:272]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 4]
```diff
@@ -334,7 +340,8 @@ public void checkTypeParameterNullnessForFunctionReturnType(
    * @param lhsType type for the lhs of the assignment
    * @param rhsType type for the rhs of the assignment
    */
-  private boolean compareNullabilityAnnotations(Type.ClassType lhsType, Type.ClassType rhsType) {
+  private static boolean compareNullabilityAnnotations(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:343]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 5]
```diff
@@ -395,7 +402,8 @@ private boolean compareNullabilityAnnotations(Type.ClassType lhsType, Type.Class
    * @param tree A parameterized typed tree for which we need class type with preserved annotations.
    * @return A Type with preserved annotations.
    */
-  private Type.ClassType typeWithPreservedAnnotations(ParameterizedTypeTree tree) {
+  private static Type.ClassType typeWithPreservedAnnotations(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:405]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 6]
```diff
@@ -460,31 +469,32 @@ private Type.ClassType typeWithPreservedAnnotations(ParameterizedTypeTree tree)
    *
    * @param tree A conditional expression tree to check
    */
-  public void checkTypeParameterNullnessForConditionalExpression(ConditionalExpressionTree tree) {
-    if (!config.isJSpecifyMode()) {
+  public static void checkTypeParameterNullnessForConditionalExpression(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:472]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 7]
```diff
@@ -500,11 +510,13 @@ public void checkTypeParameterNullnessForConditionalExpression(ConditionalExpres
    * @param actualParams the actual parameters
    * @param isVarArgs true if the call is to a varargs method
    */
-  public void compareGenericTypeParameterNullabilityForCall(
+  public static void compareGenericTypeParameterNullabilityForCall(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:513]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 8]
```diff
@@ -555,18 +567,24 @@ public void compareGenericTypeParameterNullabilityForCall(
    * @param overridingMethod A symbol of the overriding method
    * @param overriddenMethod A symbol of the overridden method
    */
-  public void checkTypeParameterNullnessForMethodOverriding(
-      MethodTree tree, Symbol.MethodSymbol overridingMethod, Symbol.MethodSymbol overriddenMethod) {
-    if (!config.isJSpecifyMode()) {
+  public static void checkTypeParameterNullnessForMethodOverriding(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:570]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 9]
```diff
@@ -686,15 +704,20 @@ public static Nullness getGenericReturnNullnessAtInvocation(
    * @return Nullness of parameter at {@code paramIndex}, or {@code NONNULL} if the call does not
    *     invoke an instance method
    */
-  public Nullness getGenericParameterNullnessAtInvocation(
-      int paramIndex, Symbol.MethodSymbol invokedMethodSymbol, MethodInvocationTree tree) {
+  public static Nullness getGenericParameterNullnessAtInvocation(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:707]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 10]
```diff
@@ -726,8 +749,12 @@ public Nullness getGenericParameterNullnessAtInvocation(
    * @return nullability of the relevant parameter type of {@code method} in the context of {@code
    *     enclosingType}
    */
-  public Nullness getGenericMethodParameterNullness(
-      int parameterIndex, Symbol.MethodSymbol method, Type enclosingType) {
+  public static Nullness getGenericMethodParameterNullness(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:752]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 11]
```diff
@@ -741,8 +768,8 @@ public Nullness getGenericMethodParameterNullness(
    * @param tree tree for overriding method
    * @param overriddenMethodType type of the overridden method
    */
-  private void checkTypeParameterNullnessForOverridingMethodParameterType(
-      MethodTree tree, Type overriddenMethodType) {
+  private static void checkTypeParameterNullnessForOverridingMethodParameterType(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:771]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 12]
```diff
@@ -771,18 +801,20 @@ private void checkTypeParameterNullnessForOverridingMethodParameterType(
    * @param tree tree for overriding method
    * @param overriddenMethodType type of the overridden method
    */
-  private void checkTypeParameterNullnessForOverridingMethodReturnType(
-      MethodTree tree, Type overriddenMethodType) {
+  private static void checkTypeParameterNullnessForOverridingMethodReturnType(
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:804]: Javadoc needs to be updated with the new `@param`s.

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java [hunk 13]
```diff
@@ -2414,6 +2413,10 @@ public ErrorBuilder getErrorBuilder() {
     return errorBuilder;
   }
 
+  public Config getConfig() {
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/NullAway.java:2416]: Sorry if this is unbelievably pedantic but: public method -> javadoc 😅 (Also, as a getter, maybe worth moving further up within this file? Closer to the field unless that breaks GJF rules?)

**Review by lazaroclapp — COMMENTED** [review]
Javadoc nits only, but quite a few of them...

**Comment by msridhar** [comment]
Thanks for helping keep our docs in a decent shape 🙂 @lazaroclapp this is ready for another look when you can

**Review by lazaroclapp — APPROVED** [review]
LGTM!

**Pull request merged**